pkg/resources/kafka/configmap.go

```diff
@@ -275,3 +280,113 @@ func (r Reconciler) generateBrokerConfig(id int32, brokerConfig *v1beta1.BrokerC
 
 	return strings.Join(completeConfig, "\n")
 }
+
+func (r *Reconciler) updateBrokerStatus(log logr.Logger, desired *corev1.ConfigMap) error {
```

Comment:
I think this should be inside the k8sreconciler method. We already identified the broker related configmaps during an update, all these things can be simplified with ~4 lines of code if we place this inside that method. Moreover, it removes the k8s client usage from a resource related file. If we check all other resource related file their scope is to create an object only, without any k8s interaction.

Reply:
Agreed. Resolve this by putting that code back, and moving config related functions to `common.go` in `kafka/util` package to prevent import cycle being created.